crates/router/src/core/payment_methods/cards.rs

```diff
@@ -5704,6 +5734,24 @@ pub async fn delete_payment_method(
         )
         .await?;
 
+        if let Some(network_token_ref_id) = key.network_token_requestor_reference_id {
+            let resp = network_tokenization::delete_network_token_from_locker_and_token_service(
+                &state,
+                &key.customer_id,
+                &key.merchant_id,
+                key.payment_method_id.clone(),
+                key.network_token_locker_id,
+                network_token_ref_id,
+            )
+            .await?;
+
+            if resp.status == "Ok" {
+                logger::info!("Token From locker deleted Successfully!");
+            } else {
+                logger::error!("Error: Deleting Token From Locker!\n{:#?}", resp);
```

Comment:
We should add the task to process tracker if the delete failed.
Add metrics to track this issue.
If we planned to take this up in next PR create respective issue and a comment say what needs to handle and respective ticket here.